mtags/src/main/scala/scala/meta/internal/pc/OverrideCompletions.scala

```diff
@@ -0,0 +1,461 @@
+
+    lastVisistedParentTrees match {
+
+      // class Foo extends Bar {}
+      // ~~~~~~~~~~~~~~~~~~~~~~~~
+      case (c: ClassDef) :: _ =>
+        val t = c.impl
+        val typed = typedTreeAt(t.pos)
+        implementAll(
+          typed,
+          inferEditPosition(text, t).toLSP,
+          t,
+          text,
+          true,
+          _ => true
+        )
+
+      // new Foo {}
+      //     ~~~~~~
+      case (_: Ident) ::
+            (t: Template) :: _ =>
+        val typed = typedTreeAt(t.pos)
+        implementAll(
+          typed,
+          inferEditPosition(text, t).toLSP,
+          t,
+          text,
+          true,
+          _ => true
+        )
+
+      // new Foo[T] {}
+      //     ~~~~~~~~~
+      case (_: Ident) ::
+            (_: AppliedTypeTree) ::
+            (t: Template) :: _ =>
+        val typed = typedTreeAt(t.pos)
+        implementAll(
+          typed,
+          inferEditPosition(text, t).toLSP,
+          t,
+          text,
+          true,
+          _ => true
+        )
+
+      case _ =>
+        Nil
+    }
+  }
+
+  /**
+   * Get text edits for an `override def` completion to implement methods from the supertype.
+   *
+   * @param typed the typed tree: template for the class/object we are implementing.
+   * @param range the position to fill the completions.
+   * @param t the enclosing template for the class/object we are implementing.
+   * @param text the text of the original source code.
+   * @param shouldAddOverrideKwd if it's true, completion add `override` for each methods.
+   * @param isCandidate the determination of whether the symbol will be a possible completion item.
+   * @return the list of TextEdit of both method implementations and auto imports.
+   */
+  private def implementAll(
+      typed: Tree,
+      range: l.Range,
+      t: Template,
+      text: String,
+      shouldAddOverrideKwd: Boolean,
+      isCandidate: Symbol => Boolean
+  ): List[l.TextEdit] = {
+    val overrideMembers = getMembers(
+      typed,
+      range,
+      t.pos,
+      text,
+      true,
+      false,
+      isCandidate
+    )
+
+    val allAbstractMembers = overrideMembers
+      .filter(_.sym.isAbstract)
+
+    val (allAbstractEdits, allAbstractImports) = toEdits(allAbstractMembers)
+
+    if (allAbstractEdits.length > 0) {
+
+      // infer necessary indent
+      //
+      // |object Test {
+      // |    class Foo extends Bar {} // inferred to 4
+      // |}
+      val lineStart = t.pos.source.lineToOffset(t.pos.line - 1)
+      val necessaryIndent = inferIndent(lineStart, text)
+
+      // infer indent for implementations
+      // if there's declaration in the class/object, follow its indent.
+      // otherwise the indent default to 2
+      val indent = typed.tpe.decls
+        .filter(sym =>
+          !sym.isSynthetic &&
+            !sym.isPrimaryConstructor &&
+            sym.pos.line != t.pos.line // filter out explicit primary constructor `class Foo(x: Int) ...`
+        )
```

Comment:
Find the explicit declaration in the class/object so that we can infer the indent from them.